crates/astria-composer/src/searcher/mod.rs

```diff
@@ -150,16 +154,29 @@ impl Searcher {
     /// + failed to connect to the eth RPC server;
     /// + failed to construct a sequencer clinet
     pub(super) async fn from_config(cfg: &Config) -> eyre::Result<Self> {
+        use rollup::Rollup;
+        let mut rollups = cfg
```

Comment:
I feel like this should happen in `Config::from_env()`. The `Config` object should have a `Vec<Rollup>` field, which is what the `Searcher` would use in its constructor.

This way `Config` is validated when it is created, and there isn't a `Config` object that's passed around with an invalid value until it reaches the `Searcher`.

Reply:
Will do

Reply:
Because `Config` is a public interface and because all of its fields are public this would require making `Rollup` public, which I don't want to do.

I am going to leave argument as `String`. Another reason to not parse it at config construction is to not worry about serializing config: it is good practice to leave the arguments as-is (or ensure they are serialized to exactly the same form as the input) to give visual feedback to the user that the config was received as provided.

The `SocketAddr` impl ensures that the latter is the case.